Makefile

```diff
@@ -1,67 +1,36 @@
 BINDIR ?= /usr/bin
 SBINDIR ?= /usr/sbin
-ETCDIR ?= /etc
 MANDIR ?= /usr/share/man/man1
-all: build doc
 
-.PHONY: test install uninstall clean
+.PHONY: build release install uninstall clean test doc reindent
 
-export OCAMLRUNPARAM=b
-J=4
+build:
+	jbuilder build @networkd/man @install --dev
 
-setup.bin: setup.ml
-	@ocamlopt.opt -o $@ $< || ocamlopt -o $@ $< || ocamlc -o $@ $<
-	@rm -f setup.cmx setup.cmi setup.o setup.cmo
-
-setup.data: setup.bin
-	@./setup.bin -configure --enable-tests
-
-build: setup.data setup.bin networkd/version.ml
-	@./setup.bin -build -j $(J)
-	mv networkd.native xcp-networkd
-	./xcp-networkd --help=groff > xcp-networkd.1
-
-networkd/version.ml: VERSION
-	echo "let version = \"$(shell cat VERSION)\"" > networkd/version.ml
-
-doc: setup.data setup.bin
-	@./setup.bin -doc -j $(J)
-
-test: setup.bin build
-	@./setup.bin -test
+release:
+	jbuilder build @install
```

Comment:
Change this to 
```
jbuilder build @install @networkd/man
```
and make this the default make target.

The `--dev` build could fail if there are warnings, we should not silence them and fix the one that can be fixed, but then build the product without the `--dev` flag

Reply:
I will update for most comments. However, for the library split up thing, I would like to do it separately, tracked by another ticket. In that case, the jbuild porting stuff will not need to wait.